### ISSUE TITLE
KPI - User across Date Range

### ISSUE THREAD

**Issue by DynamikArray** [DynamikArray/Ciima#78]
Show stats for users across Date Ranges   Last 1, 5, 7, 14, 30
Item Quantities against Stock Value,  Stock Value Aggregrates
Total New Items, Total Item Updates.

Maybe we can figure out time spent or something too "roughly".... choose a time gap to ignore as inactive, add up start to end times, etc ???

**Comment by DynamikArray** [comment]
```
SELECT
	DATE(l.created_date) AS createdDate,
	u.username,
	l.action AS auditAction,
	count(l.resource_id) AS recordCount
FROM
	slc_audit_log l	
LEFT JOIN 
	slc_users u
ON (l.user_id = u.id )
WHERE (
	(l.action = 'CREATE_DRAFT')
	OR	
	(l.action = 'UPDATE_LOCATION_FIELD')
)
GROUP BY 
	DATE(l.created_date),
	u.username,
	l.action
```

**Comment by DynamikArray** [comment]
CREATE DRAFT METRICS
```
SELECT
	DATE(l.created_date) AS createdDate,
	u.username,
	l.action AS auditAction,
	count(l.resource_id) AS totalItems,
	round(sum(d.price)) AS totalPrice,
   ROUND( sum(d.price) / count(l.resource_id) ) AS avgPrice
FROM
	slc_audit_log l	
LEFT JOIN 
	slc_users u
ON l.user_id = u.id 
LEFT JOIN
	slc_drafts d
ON 
	l.resource_id = d.id
WHERE 
(
	(l.action = 'CREATE_DRAFT')	
)
GROUP BY 
	DATE(l.created_date),
	u.username,
	l.action
ORDER BY 
	created_date desc
```

**Comment by DynamikArray** [comment]
UPDATE_LOCATION_FIELD

```
SELECT 
	stats.createdDate,
	stats.username,
	stats.auditAction,
	COUNT(stats.price) AS totalItems,
	ROUND(SUM(stats.price)) AS totalPrice,	
	ROUND ( SUM(stats.price) / COUNT(stats.price) ) AS avgCost	
FROM 
(SELECT
	DATE(l.created_date) AS createdDate,
	u.username,
	l.action AS auditAction,
	sum(d.price) AS price
FROM
	slc_audit_log l	
LEFT JOIN 
	slc_users u
ON l.user_id = u.id 
LEFT JOIN
	slc_drafts d
ON 
	l.resource_id = d.stockItemId
WHERE 
(
	(l.action = 'UPDATE_LOCATION_FIELD')	
)
GROUP BY
	d.stockItemId,	
	u.username
ORDER BY 
	DATE(l.created_date) DESC
) stats
GROUP BY 
	stats.createdDate	
ORDER BY 
	stats.createdDate	
```

**Comment by DynamikArray** [comment]
STARTED:

https://github.com/DynamikArray/Ciima/commit/aaae398a7958c986191c7f9b96975ef0e602fea7

**Comment by DynamikArray** [comment]
Closed in 4e50cf3e464e8af0b117b795b74443729b0e34ab

**Comment by DynamikArray** [comment]
Resolved KPI Grouping Issues
